internal/controller/grafanaorganization_controller.go

```diff
@@ -129,22 +134,60 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 	}
 
 	// Update CR status if anything was changed
-	if organization.ID != grafanaOrganization.Status.OrgID {
-		grafanaOrganization.Status.OrgID = organization.ID
+	if err = r.updateOrgStatus(ctx, grafanaOrganization, organization.ID); err != nil {
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	err = r.configureGrafana(ctx)
+	if err != nil {
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r GrafanaOrganizationReconciler) updateOrgStatus(ctx context.Context, grafanaOrganization *v1alpha1.GrafanaOrganization, orgID int64) error {
+	logger := log.FromContext(ctx)
+
+	// Update orgID in the CR's satus
+	if grafanaOrganization.Status.OrgID != orgID {
+		log.Log.Info("updating orgID in the org's status")
+		grafanaOrganization.Status.OrgID = orgID
 
-		if err = r.Status().Update(ctx, grafanaOrganization); err != nil {
+		if err := r.Status().Update(ctx, grafanaOrganization); err != nil {
 			logger.Error(err, "failed to update the status")
-			return ctrl.Result{}, errors.WithStack(err)
+			return errors.WithStack(err)
 		}
 	}
 
-	// TODO add datasources for the organization.
+	// Update the datasources in the CR's status
+	log.Log.Info("updating dataSources in the org's status")
+	var datasources []v1alpha1.DataSources
 
-	err = r.configureGrafana(ctx)
+	lokiDatasourceID, err := grafana.GetDatasourceID(ctx, r.GrafanaAPI, "Loki")
```

Comment:
The goal here is to create a datasource in each organization though :D

Reply:
Well that's what's being done if I'm not mistaken. The status update happens whenever the reconcileCreate method is called so all org CRs will have the datasources in their status.

Reply:
Where is the code to create the datasources exactly?

Reply:
All I see here is you're setting the status to be the existing datasoruce but this datasource exists only in the shared org today not in the others :)

Reply:
Oh ok so I think I misunderstood something : in my understanding we needed to add existing datasources to the orgs, not create new ones.
But if we actually need to create datasources for each org, then let's do it :)

Reply:
Datasources are not shared for each org so this is how we will configure the tenant :)